internal/hcs/watcher.go

```diff
@@ -17,14 +18,16 @@ import (
 // Usage is:
 //
 // completed := false
-// go syscallWatcher("some description", &completed)
+// go syscallWatcher(context, &completed)
 // <syscall>
 // completed = true
 //
-func syscallWatcher(description string, syscallCompleted *bool) {
+func syscallWatcher(context logrus.Fields, syscallCompleted *bool) {
```

Comment:
I wonder whether this should take a *logrus.Entry instead of a logrus.Fields, and the common field routine should return an entry instead of the field map. 

This is a minor difference, but it does remove one allocation. Also, it gives us flexibility to choose the non-standard logger without having to change this function.